types/ramda/index.d.ts

```diff
@@ -888,6 +888,7 @@ export function identity<T>(a: T): T;
  */
 // tslint:disable:max-line-length
 export function ifElse<TArgs extends any[], TOnTrueResult, TOnFalseResult>(fn: (...args: TArgs) => boolean, onTrue: (...args: TArgs) => TOnTrueResult, onFalse: (...args: TArgs) => TOnFalseResult): (...args: TArgs) => TOnTrueResult | TOnFalseResult;
+export function ifElse<TArgs extends any[], U extends any, N extends U, TOnTrueResult, TOnFalseResult>(fn: (u: U) => u is N, onTrue: (n: N, ...args: TArgs) => TOnTrueResult, onFalse: (n: Exclude<U, N>, ...args: TArgs) => TOnFalseResult): (u: U, ...args: TArgs) => TOnTrueResult | TOnFalseResult;
```

Comment:
Is `TArgs` still needed?  Is it just there to support the case where the predicate takes fewer arguments than `onTrue` and `onFalse`?

What does the `extends any` in `U extends any` do?